aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -0,0 +1,614 @@
+
+Register health check services with `AddHealthChecks` in `Startup.ConfigureServices`. Call Health Check Middleware in the app processing pipeline in `Startup.Configure`.
+
+In the sample app, the health check endpoint is created at `/health` (*BasicStartup.cs*):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/BasicStartup.cs?name=snippet1&highlight=5,10)]
+
+To run the basic configuration scenario using the sample app, execute the following command from the project's folder in a command shell:
+
+```console
+dotnet run --scenario basic
+```
+
+### Docker example
+
+[Docker](xref:host-and-deploy/docker/index) offers a built-in `HEALTHCHECK` directive that can be used to check the status of an app that uses the basic health check configuration:
+
+```
+HEALTHCHECK CMD curl --fail http://localhost:5000/health || exit
+```
+
+## Create health checks
+
+Health checks are created by implementing the `IHealthCheck` interface. The `IHealthCheck.CheckHealthAsync` method returns a `Task<HealthCheckResult>` that indicates the health as `Healthy`, `Degraded`, or `Unhealthy`. The result is written as a plaintext response with a configurable status code (configuration is described in the [Health check options](#health-check-options) section). `HealthCheckResult` can also return optional key-value pairs.
+
+### Example health check
+
+The following `ExampleHealthCheck` class demonstrates the layout of a health check:
+
+```csharp
+public class ExampleHealthCheck : IHealthCheck
+{
+    public ExampleHealthCheck()
+    {
+        // Use dependency injection (DI) to supply any required services to the
+        // health check.
+    }
+
+    public Task<HealthCheckResult> CheckHealthAsync(
+        HealthCheckContext context, 
+        CancellationToken cancellationToken = default(CancellationToken))
+    {
+        // Execute health check logic here. This example sets a dummy
+        // variable to true.
+        var healthCheckResultHealthy = true;
+
+        if (healthCheckResultHealthy)
+        {
+            return Task.FromResult(
+                HealthCheckResult.Healthy("The check indicates a healthy result."));
+        }
+
+        return Task.FromResult(
+            HealthCheckResult.Unhealthy("The check indicates an unhealthy result."));
+    }
+}
+```
+
+### Register health check services
+
+The `ExampleHealthCheck` type is added to health check services with `AddCheck`:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHealthChecks()
+        .AddCheck<ExampleHealthCheck>("example_health_check");
+}
+```
+
+An `AddCheck` overload provides the opportunity to set the failure status (`HealthStatus`) that should be reported when the health check reports a failure. If the failure status is set to `null` (default), `HealthStatus.Unhealthy` is reported.
```

Comment:
Seeing these two examples next to each other is a little confusing. The failure status that's passed in is a recommendation that the check **should** return it. This is a feature for library authors, and requires the library to honor it.. This should be called out somwehre with a note.

Reply:
Thanks @unaizorrilla ... I'll get it on the next commit.